kernel/src/engine/arrow_utils.rs

```diff
@@ -459,6 +459,52 @@ fn get_indices(
     ))
 }
 
+/// Contains information about a StructField
+/// # Fields:
+/// * `0` - The index of the struct field in its struct
+/// * `1` - A reference to the struct field's name
+/// * `2` - A reference to the struct field
+type FieldInfo<'k> = (usize, &'k String, &'k StructField);
+
+/// Constructs an iterator where each parquet Field in `fields` is matched
+/// with a a kernel `FieldInfo` reperesenting a StructField.
+///
+/// This returns an iterator of tuples. Each tuple contains the parquet field's ordinal
+/// and the matching `FieldInfo` if present.
+fn get_field_infos<'k, 'p>(
+    requested_schema: &'k StructType,
+    fields: &'p Fields,
+) -> impl Iterator<Item = (usize, &'p Arc<Field>, Option<FieldInfo<'k>>)> {
+    // Construct a map from the field id to its StructField
+    let field_id_to_name: HashMap<i64, &String> = requested_schema
+        .fields()
+        .filter_map(
+            |field| match field.get_config_value(&ColumnMetadataKey::ParquetFieldId) {
+                Some(MetadataValue::Number(fid)) => Some((*fid, field.name())),
+                _ => None,
+            },
```

Comment:
maybe slightly cleaner? i'm okay if you like what you have better
```suggestion
            |field| if let Some(MetadataValue::Number(fid)) = field.get_config_value(&ColumnMetadataKey::ParquetFieldId) {
              Some((*fid, field.name()))
            } else{
              None
            }
```

Reply:
hmm yeah I lean more towards the match just cuz it's shorter